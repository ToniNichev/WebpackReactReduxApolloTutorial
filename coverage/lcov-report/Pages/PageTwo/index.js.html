
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Pages/PageTwo/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> / <a href="index.html">Pages/PageTwo</a> index.js</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>6/6</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>5/5</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>6/6</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a></td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { useState } from 'react';
import styles from './styles.scss';
&nbsp;
&nbsp;
&nbsp;
function PageTwo(props) {
  const [labelData, setLabelData] = useState('no data');
&nbsp;
&nbsp;
  function setTest(val) {
    window.location.hash = val;
  }
&nbsp;
  return(&lt;div data-text={labelData} className={styles.wrapper}&gt;
          {labelData}
          &lt;hr /&gt;
          &lt;input type="text" placeholder={labelData} /&gt;
          &lt;button onClick={ () =&gt; { setTest('TEST'); } }&gt;TEST&lt;/button&gt;
          &lt;ul&gt;
            &lt;li&gt;&lt;a href="#" onClick={ () =&gt; { setLabelData('one') } }&gt;ONE&lt;/a&gt;&lt;/li&gt;
            &lt;li&gt;&lt;a href="#" onClick={ () =&gt; { setLabelData('two') } }&gt;TWO&lt;/a&gt;&lt;/li&gt;
          &lt;/ul&gt;   
        &lt;/div&gt;);
}
export default PageTwo;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Nov 02 2020 15:43:17 GMT-0500 (Eastern Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    