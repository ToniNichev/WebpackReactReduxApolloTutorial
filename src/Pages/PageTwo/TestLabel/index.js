import React from 'react';

const MOD_URL_BASE = 'apps.cnbc.com';

const TestLabel = (props) => {
  const {
    testOne,
  } = props;

  
  return(<div>
        {testOne}  
        </div>);
}
export default TestLabel;
