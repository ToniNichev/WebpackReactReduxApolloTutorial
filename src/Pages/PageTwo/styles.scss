.appWrapper {
  background: rgb(141, 141, 172);
}