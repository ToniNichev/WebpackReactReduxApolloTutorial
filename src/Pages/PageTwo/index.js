import React, { useState } from 'react';
import TestLabel from './TestLabel';
import styles from './styles.scss';

let init = false;

const MOD_URL_BASE = 'apps.cnbc.com';

const PageTwo = (props) => {
  const {
    testOne,
    testTwo
  } = props;

  const [labelData, setLabelData] = useState('no data');

  const clickTest = () => {
    setLabelData('click test');
  }

  React.useEffect(() => {
    console.log("USE EFFECT CALLED !");
    init = true;

    window.addEventListener('message', function(event) {
      console.log("Message received:", event);
      if(event.origin.indexOf(MOD_URL_BASE) === -1) {
        document.querySelector("input").value = "Message received";
      }
    });   

    const url = 'https://www.toni-develops.com/external-files/examples/sample-apis/users.json.php';

    fetch(url)
      .then((res) => res.json())
      .then((newData) => {
        console.log("@#@#@#@#@", newData[0]);

        const welcomeText = `Welcome ${newData[0].first_name}`;
        setLabelData(welcomeText);

        if( document.querySelector("input") != null ) {
          console.log("useEffect:",3);
          document.querySelector("input").style.display = "";
        }
        if(document.querySelector('[data-id]')!= null) {
          console.log("###################:", welcomeText);
          document.querySelector('[data-id]').innerText = welcomeText;
        }
        return welcomeText;
      });      
  }, [init]);


  return(<div data-text={labelData} className={styles.wrapper}>
          {labelData}
          <hr />
          <TestLabel testOne={labelData}/>
          <p><button onClick={ () => { clickTest() } }>Click Test</button></p>
          <input type="text" placeholder={labelData} readOnly value={labelData} />
          <div data-id="test1">test div</div>
          <ul>
            <li><a href="#" onClick={ () => { setLabelData('one') } }>ONE</a></li>
            <li><a href="#" onClick={ () => { setLabelData('two') } }>TWO</a></li>
          </ul>   
        </div>);
}
export default PageTwo;
