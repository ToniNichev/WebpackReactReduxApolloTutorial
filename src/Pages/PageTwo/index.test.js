import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import PageTwo from './';
import toJson from 'enzyme-to-json';


// mock fetch function
global.fetch = (url) => {
  return new Promise((resolve) => {

      resolve({
        json: () => {
          return new Promise((resolve) => {
            
            resolve([{url: url, first_name: 'Tony' } ] );            
          });
        }
      });
  })
}

// create message event
var event = new Event('message');
event.origin = "test 123";


describe('PageTwo component', () => {

  // covers: if( document.querySelector("input") != null ) { ... }
  const inputTag = document.createElement('input');
  document.body.appendChild(inputTag);

  const divTag = document.createElement('div');
  divTag.setAttribute('data-id', "test-id");
  document.body.appendChild(divTag);


  let wrapper;

  // wrapping with `act` to prevent warning messages
  it('test a links',async () => {

    await act(async () => {
      wrapper = mount(<PageTwo testOne="One" testTwo="Two"/>);
    });

    console.log("selects second a tag", wrapper.find('a').at(1).debug());
    
    wrapper.find('a').at(1).simulate('click');
    console.log(wrapper.debug());

    wrapper.find('a').at(0).simulate('click');
    console.log(wrapper.debug());
  });

  /*
  it('renders as expected',async () => {
    wrapper.find('[data-id]').text().toBe('');
  });
  */

  it('renders as expected',async () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });  

  it('click button test', async () => {
    const btn = wrapper.find('button').simulate('click');
    expect( wrapper.find('input').props().value).toBe('click test');
  });

  
  it('test mock event listener',async () => {  
    window.dispatchEvent(event);
    console.log(">>>>>", wrapper.find('[data-id]').text());
    expect( document.querySelector('[data-id]').innerText).toBe('Welcome Tony');
  });
  
  /*
  it('test mock event listener',async () => {  
    jest.spyOn(window, 'addEventListener').mockImplementationOnce((event, handler, options) => {
      console.log("#######################");
    }); 
  });
  */

  
});

