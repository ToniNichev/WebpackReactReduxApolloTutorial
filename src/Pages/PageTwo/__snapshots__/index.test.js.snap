// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTwo component renders as expected 1`] = `
<PageTwo
  testOne="One"
  testTwo="Two"
>
  <div
    data-text="one"
  >
    one
    <hr />
    <PageTwo
      testOne="one"
    >
      <div>
        one
      </div>
    </PageTwo>
    <p>
      <button
        onClick={[Function]}
      >
        Click Test
      </button>
    </p>
    <input
      placeholder="one"
      readOnly={true}
      type="text"
      value="one"
    />
    <div
      data-id="test1"
    >
      test div
    </div>
    <ul>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          ONE
        </a>
      </li>
      <li>
        <a
          href="#"
          onClick={[Function]}
        >
          TWO
        </a>
      </li>
    </ul>
  </div>
</PageTwo>
`;
